snakemake/sourcecache.py

```diff
@@ -397,34 +378,20 @@ def _do_cache(self, source_file, cache_entry):
             # as mtime.
             os.utime(cache_entry, times=(mtime, mtime))
 
-    def _open_local_or_remote(self, source_file, mode):
-        from retry import retry_call
-
-        if source_file.is_local:
-            return self._open(source_file, mode)
-        else:
-            return retry_call(
-                self._open,
-                [source_file, mode],
-                tries=3,
-                delay=3,
-                backoff=2,
-                logger=logger,
-            )
-
-    def _open(self, source_file, mode):
+    def _open(self, path_or_uri, mode):
         from smart_open import open
 
-        if isinstance(source_file, LocalGitFile):
+        if isinstance(path_or_uri, LocalGitFile):
             import git
 
             return io.BytesIO(
-                git.Repo(source_file.repo_path)
-                .git.show("{}:{}".format(source_file.ref, source_file.path))
+                git.Repo(path_or_uri.repo_path)
+                .git.show("{}:{}".format(path_or_uri.ref, path_or_uri.path))
```

Comment:
I believe these changes are not needed anymore. I had already refactored the main branch based on ideas from your PR before.